src/dhcpmon/src/dhcp_device.c

```diff
@@ -242,62 +222,158 @@ static void read_callback(int fd, short event, void *arg)
 }
 
 /**
- * @code dhcp_device_validate(counters, counters_snapshot);
+ * @code dhcp_device_is_dhcp_inactive(counters);
  *
- * @brief validate current interface counters by comparing aggregate counter with snapshot counters.
+ * @brief Check if there were no DHCP activity
+ *
+ * @param counters  current/snapshot counter
+ *
+ * @return true if there were no DHCP activity, false otherwise
+ */
+static bool dhcp_device_is_dhcp_inactive(uint64_t counters[][DHCP_DIR_COUNT][DHCP_MESSAGE_TYPE_COUNT])
+{
+    uint64_t *rx_counters = counters[DHCP_COUNTERS_CURRENT][DHCP_RX];
+    uint64_t *rx_counter_snapshot = counters[DHCP_COUNTERS_SNAPSHOT][DHCP_RX];
+
+    bool rv = true;
+    for (uint8_t i = 0; (i < monitored_msg_sz) && rv; i++) {
+        rv = rx_counters[monitored_msgs[i]] == rx_counter_snapshot[monitored_msgs[i]];
+    }
+
+    return rv;
+}
+
+/**
+ * @code dhcp_device_is_dhcp_msg_unhealthy(type, counters);
+ *
+ * @brief Check if there were no DHCP activity
+ *
+ * @param type      DHCP message type
+ * @param counters  current/snapshot counter
+ *
+ * @return true if DHCP message 'type' is transmitted,false otherwise
+ */
+static bool dhcp_device_is_dhcp_msg_unhealthy(dhcp_message_type_t type,
+                                              uint64_t counters[][DHCP_DIR_COUNT][DHCP_MESSAGE_TYPE_COUNT])
+{
+    // check if DHCP message 'type' is being relayed
+    return ((counters[DHCP_COUNTERS_CURRENT][DHCP_RX][type] >  counters[DHCP_COUNTERS_SNAPSHOT][DHCP_RX][type]) &&
+            (counters[DHCP_COUNTERS_CURRENT][DHCP_TX][type] <= counters[DHCP_COUNTERS_SNAPSHOT][DHCP_TX][type])    );
+}
+
+/**
+ * @code dhcp_device_check_positive_health(counters, counters_snapshot);
+ *
+ * @brief validate current interface counters by comparing aggregate counters with snapshot counters.
+ *
+ * @param counters  current/snapshot counter
+ *
+ * @return DHCP_MON_STATUS_HEALTHY, DHCP_MON_STATUS_UNHEALTHY, or DHCP_MON_STATUS_INDETERMINATE
+ */
+static dhcp_mon_status_t dhcp_device_check_positive_health(uint64_t counters[][DHCP_DIR_COUNT][DHCP_MESSAGE_TYPE_COUNT])
+{
+    dhcp_mon_status_t rv = DHCP_MON_STATUS_HEALTHY;
+
+    bool is_dhcp_unhealthy = false;
+    for (uint8_t i = 0; (i < monitored_msg_sz) && !is_dhcp_unhealthy; i++) {
+        is_dhcp_unhealthy = dhcp_device_is_dhcp_msg_unhealthy(monitored_msgs[i], counters);
+    }
+
+    // if we have rx DORA then we should have corresponding tx DORA (DORA being relayed)
+    if (is_dhcp_unhealthy) {
+        rv = DHCP_MON_STATUS_UNHEALTHY;
+    }
+
+    return rv;
+}
+
+/**
+ * @code dhcp_device_check_negative_health(counters);
+ *
+ * @brief validate current interface counters by comparing aggregate counters with snapshot counters.
```

Comment:
Could you please add more detail to this brief description and also for `dhcp_device_check_positive_health` above differentiate them so that it's easier to understand what "positive" and "negative" health checks are? Currently both descriptions are the same, and are also the same as that for `dhcp_device_check_health` below.